File: samples/client/petstore/javascript/src/scripts/rest/model/Tag.js



//export module
if ( typeof define === "function" && define.amd ) {     
	define('Tag', ['jquery'], 
		function($) {
        return Tag;
	 });
}


var Tag = function Tag() { 
  	var self = this;
  	
  	/**
	  * datatype: Long
	  **/
	  self.id = null;
	 
  	/**
	  * datatype: String
	  **/
	  self.name = null;
	 
  
  	self.constructFromObject = function(data) {
	  	
	  	self.id = data.id;
	  	
	  	self.name = data.name;
	  	
	}
  
  
  /**
   * @return {Long}
   **/
  self.getId = function() {
    return self.id;
  }
  
  /**
   * @param {Long} id
   **/
  self.setId = function (id) {
    self.id = id;
  }
  
  /**
   * @return {String}
   **/
  self.getName = function() {
    return self.name;
  }
  
  /**
   * @param {String} name
   **/
  self.setName = function (name) {
    self.name = name;
  }
  

  self.toJson = function () {
  	return JSON.stringify(self);
  }
}

File: samples/client/petstore/cpp-tiny/lib/Models/Tag.cpp


#include "Tag.h"

using namespace Tiny;

Tag::Tag()
{
	id = long(0);
	name = std::string();
}

Tag::Tag(std::string jsonString)
{
	this->fromJson(jsonString);
}

Tag::~Tag()
{

}

void
Tag::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *idKey = "id";

    if(object.has_key(idKey))
    {
        bourne::json value = object[idKey];



        jsonToValue(&id, value, "long");


    }

    const char *nameKey = "name";

    if(object.has_key(nameKey))
    {
        bourne::json value = object[nameKey];



        jsonToValue(&name, value, "std::string");


    }


}

bourne::json
Tag::toJson()
{
    bourne::json object = bourne::json::object();





    object["id"] = getId();






    object["name"] = getName();



    return object;

}

long
Tag::getId()
{
	return id;
}

void
Tag::setId(long  id)
{
	this->id = id;
}

std::string
Tag::getName()
{
	return name;
}

void
Tag::setName(std::string  name)
{
	this->name = name;
}



